### ISSUE TITLE
Error decoding JSON: json: cannot unmarshal array into Go struct field ContainerDefinition.DockerLabels of type string

### ISSUE THREAD

**Issue by nodesocket** [terraform-providers/terraform-provider-aws#10108]
Not sure if this belongs here in the AWS provider or core Terraform. Using `aws_ecs_task_definition` and the attribute `container_definitions`. Providing additional JSON to my task definition of:

```
 "dockerLabels": {
        "com.datadoghq.ad.check_names": [
            "apache"
        ],
        "com.datadoghq.ad.init_configs": [
            {}
        ],
        "com.datadoghq.ad.instances": [[
            {
                "apache_status_url": "http://%%host%%/server-status?auto"
            }
        ]]
    }
```

Breaks with:

```
Error: module.ecs_task_definiton.aws_ecs_task_definition.definition: ECS Task Definition container_definitions is invalid: Error decoding JSON: json: cannot unmarshal array into Go struct field ContainerDefinition.DockerLabels of type string
```

This howerver is valid JSON. I further narrowed this bug down to just:

```
 "dockerLabels": {
    "foo": []
}
```

Which also throws the same error about error decoding JSON. There seems to be bug in the unmarshalling of JSON related to `map => key: array`.

Running Terraform:

```
Terraform v0.11.13
+ provider.aws v2.28.1
+ provider.random v2.2.0
```

**Comment by Millerborn** [comment]
Also came across this issue

**Comment by dileepfrog** [comment]
@nodesocket @Millerborn Did you guys happen to find a workaround to adding Docker labels to the ECS containers?

**Comment by nodesocket** [comment]
@dileepfrog I still don't have a fix. Not sure if this got fixed in Terraform v0.12.X, but I can't upgrade.

**Comment by bflad** [comment]
Hi folks 👋 According to the [ECS API Reference](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_RegisterTaskDefinition.html), the API is expecting Docker labels to be defined as a map with string keys and string values:

```
{
   "containerDefinitions": [ 
      { 
...
         "dockerLabels": { 
            "string" : "string" 
         },
```

This representation is then enforced by [AWS Go SDK](https://docs.aws.amazon.com/sdk-for-go/api/service/ecs/#ContainerDefinition.SetDockerLabels), which is the cause of the JSON error.

Since differing container definition syntax this does not appear to be supported by the ECS API and AWS Go SDK, I'm going to close this issue. Hope this helps though.
